### PR TITLE
Add manifests for product construction

### PR DESCRIPTION
**Pull request by mmitche**
Will set up the input parameters (commit/build id, etc.) as part of the build definition.

### CODE REVIEW

File: PublishToBlob.proj [hunk 1]
```diff
@@ -24,12 +24,23 @@
     <PushToBlobFeed ExpectedFeedUrl="$(ExpectedFeedUrl)"
                     AccountKey="$(AccountKey)"
                     ItemsToPush="@(NuGetPackages)"
-                    Overwrite="$(PublishOverwrite)" />
+                    Overwrite="$(PublishOverwrite)"
+                    ManifestBranch="$(ManifestBranch)"
```
Comment (brettfo) [inline PublishToBlob.proj]: How do these `$(Manifest*)` variables make it into the build?  Are they environment variables or does the build definition need to be modified to pass them in?

Comment (mmitche) [reply PublishToBlob.proj]: Modify the build definition.  Mainly pass the source branch/hash.

**Comment by mmitche** [comment]
@brettfo Good to merge?

**Comment by brettfo** [comment]
This should target `master`; our `dev15.6` branch is in escrow.

**Comment by mmitche** [comment]
@brettfo We have to have this for 2.1 prodcon...isn't 15.6 the 2.1 branch?

**Comment by brettfo** [comment]
Ok, 15.6 is fine.  I'll merge now.

**Pull request merged**